File: algorithms/code/leetcode/lc190_reverse_bits/lc190_reverse_bits.py
# https://leetcode.com/problems/reverse-bits/


class Solution:
    def reverseBits(self, n: int) -> int:
        r = 0
        for _ in range(0, 32):
            r <<= 1
            r |= n & 1
            n >>= 1
        return r


class TestCode:
    def test_43261596(self) -> None:
        assert Solution().reverseBits(43261596) == 964176192

File: algorithms/code/leetcode/lc190_reverse_bits/lc190_reverse_bits.go
// Package lc190 implements https://leetcode.com/problems/reverse-bits/
package lc190

func reverseBits(n int) int {
	r := 0
	for range 32 {
		r <<= 1
		r |= n & 1
		n >>= 1
	}
	return r
}
